crates/sui-types/src/messages_checkpoint.rs

```diff
@@ -119,10 +124,134 @@ impl Default for ECMHLiveObjectSetDigest {
     }
 }
 
+/// CheckpointArtifact is a type that represents various artifacts of a checkpoint.
+/// We hash all the artifacts together to get the checkpoint artifacts digest
+/// that is included in the checkpoint summary.
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub enum CheckpointArtifact {
+    ObjectStates(BTreeMap<ObjectID, (SequenceNumber, ObjectDigest)>),
+    // In the future, we can add more artifacts e.g., execution digests, events, etc.
+}
+
+impl CheckpointArtifact {
+    pub fn digest(&self) -> SuiResult<Digest> {
+        match self {
+            Self::ObjectStates(object_states) => {
+                let tree = MerkleTree::<Blake2b256>::build_from_unserialized(
+                    object_states
+                        .iter()
+                        .map(|(id, (seq, digest))| (id, seq, digest)),
+                )
+                .map_err(|e| SuiError::GenericAuthorityError {
+                    error: format!("Failed to build Merkle tree: {}", e),
+                })?;
+                let root = tree.root().bytes();
+                Ok(Digest::new(root))
+            }
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct CheckpointArtifacts {
+    artifacts: BTreeSet<CheckpointArtifact>,
+}
+
+#[derive(Debug)]
+pub struct CheckpointArtifactDigests {
+    pub digests: Vec<Digest>,
+}
```

Comment:
Whats the purpose of this? the value that is actually committed to in the checkpoint is only a single Digest. I see that the CheckpointArtifact type is an enum that right now only tracks object states but once a new variant is added there i don't think theres a way to determine which digest corresponds which variant

Reply:
Mostly i'm trying to figure out, given we have the list of Checkpoint Commitment digests (which is a list of enums) how does that factor in to this

Reply:
Thanks for catching. `CheckpointArtifactDigests` is indeed unnecessary: I just removed it and moved the logic into `CheckpointArtifacts::digest()`. 

As I read your question, another thought came up: should we introduce `CheckpointArtifactDigest` in order to avoid overloading `Digest`? I don't have any strong preference.. but let me know if you think we should have it.
 
```impl CheckpointArtifact {
    pub fn digest(&self) -> SuiResult<Digest> {
```
with 
```impl CheckpointArtifact {
    pub fn digest(&self) -> SuiResult<CheckpointArtifactDigest> {
```